src/main/java/com/powsybl/openloadflow/ac/outerloop/tap/GeneratorVoltageControlManager.java

```diff
@@ -0,0 +1,105 @@
+    }
+
+    private static double calculateMaxControlledNominalVoltage(LfNetwork network) {
+        double maxControlledNominalVoltage = Double.MIN_VALUE;
+        for (LfBus bus : network.getBuses()) {
+            if (!bus.isDisabled()
+                    && bus.isTransformerVoltageControlled()
+                    && isTransformerVoltageControlsValidForMaxControlledNominalVoltageCalculation(bus.getTransformerVoltageControl().orElse(null))) {
+                maxControlledNominalVoltage = Math.max(maxControlledNominalVoltage, bus.getNominalV());
+            }
+        }
+        return maxControlledNominalVoltage;
+    }
+
+    private static boolean isTransformerVoltageControlsValidForMaxControlledNominalVoltageCalculation(TransformerVoltageControl transformerVoltageControl) {
+        // are removed from this automatic algorithm the transformer voltage control that are between two nominal
+        // voltages equivalents.
+        if (transformerVoltageControl != null) {
+            for (LfBranch branch : transformerVoltageControl.getControllerElements()) {
+                if (!branch.isConnectedAtBothSides()
+                        || branch.getBus1().getNominalV() == branch.getBus2().getNominalV()) {
+                    return false;
+                }
+            }
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     *  Disables the voltage control of generators with nominal voltage below the limit
+     * @return true if at least one generator is disabled. False if the model is not modified
+     */
+    public boolean stopTensionControlBelowLimit(LfNetwork network) {
+
+        boolean result = false;
+
+        for (LfBus bus : network.getControlledBuses(VoltageControl.Type.GENERATOR)) {
+            if (bus.getNominalV() < minNominalVoltageLimit) {
+                var voltageControl = bus.getGeneratorVoltageControl().orElseThrow();
+                for (LfBus controllerBus : voltageControl.getMergedControllerElements()) {
+                    if (controllerBus.isGeneratorVoltageControlEnabled() && !isBusBehindVeryHighVoltageTransfo(controllerBus, minNominalVoltageLimit)) {
+                        controllerBus.setGenerationTargetQ(controllerBus.getQ().eval());
+                        controllerBus.setGeneratorVoltageControlEnabled(false);
+                        busesWithVoltageControlDisabled.add(controllerBus);
+                        result = true;
+                    }
+                }
+            }
+        }
+        return result;
+    }
+
+    public void restartGeneratorTensionControl() {
+        for (LfBus controllerBus : busesWithVoltageControlDisabled) {
+            controllerBus.setGenerationTargetQ(0);
+            controllerBus.setGeneratorVoltageControlEnabled(true);
+        }
+    }
+
+    private boolean isBusBehindVeryHighVoltageTransfo(LfBus bus, double limit) {
```

Comment:
This is not generic enough, especially the detection of step-up transformers. It is only for local voltage control? Meaning: the generator is controlling voltage at it connection bus and "after" the connection bus, we have a step-up transformer?

Reply:
And a load tap changing capabilities to false.